### ISSUE TITLE
Doesn't work for custom lib.rs filename

### ISSUE THREAD

**Issue by ogoffart** [rust-mobile/xbuild#161]
I'm using a custom filename in my Cargo.toml like so:

```
[lib]
crate-type = ["cdylib"]
path = "custom.rs"
```

Then, if i run `x build --arch arm64 --platform android` , i'd have an output like so:

```
[1/3] Fetch precompiled artifacts
info: component 'rust-std' for target 'aarch64-linux-android' is up to date
[1/3] Fetch precompiled artifacts [112ms]
[2/3] Build rust `my_app` [SKIPPED]
[3/3] Create apk [429ms]
```

It wouldn't build the app and just print `[SKIPPED]`.  
It took me a while to realize this is because I'm using a custom path and there is no src/lib.rs
Probably due to this check.
https://github.com/rust-mobile/xbuild/blob/268939a99a50d7927bce3e343e5d2ed32eb8b30e/xbuild/src/command/build.rs#L30

I'm not quite sure why this check is there, but I would recommend checking the existence of a `[lib]` section in Cargo.toml as well.

**Comment by MarijnS95** [comment]
@ogoffart indeed! I've previously solved this for `cargo-apk` (""deprecated"") in https://github.com/rust-mobile/cargo-subcommand/pull/17 (including renames via `name = ""`), and am now mostly unhappy that `xbuild` has mostly copy-pasted older implementations rather than reusing that crate (or any other crate that parses/reflects a workspace). To the point that I don't feel like typing up the solutions for the second time :)
